client/app/bundles/course/assessment/pages/AssessmentShow/prompts/ConvertMcqMrqPrompt.tsx

```diff
@@ -0,0 +1,132 @@
+import { useState } from 'react';
+import { toast } from 'react-toastify';
+import { East } from '@mui/icons-material';
+import { Alert, Chip, Typography } from '@mui/material';
+import { McqData } from 'types/course/assessment/assessments';
+
+import Prompt, { PromptText } from 'lib/components/core/dialogs/Prompt';
+import useTranslation from 'lib/hooks/useTranslation';
+
+import { convertMcqMrq } from '../../../actions';
+import translations from '../../../translations';
+
+interface ConvertMcqMrqPromptProps {
+  for: McqData;
+  onClose: () => void;
+  onConvertComplete: (data: McqData) => void;
+  open: boolean;
+}
+
+const ConvertMcqMrqPrompt = (props: ConvertMcqMrqPromptProps): JSX.Element => {
+  const { for: question } = props;
+  const { t } = useTranslation();
+  const [converting, setConverting] = useState(false);
+
+  const convert = (unsubmit: boolean, convertUrl?: string) => () => {
+    if (!convertUrl)
+      throw new Error(
```

Comment:
This is an assertion; it's just that using `console.assert` would trip ESLint, so I threw an `Error` instance instead.

The idea is that _in production code_, `convertUrl` should _never, ever_ be _not a string_ when this function is called. But we cannot restrict `convertUrl: string`, because the value is sent from back-end, and it has no contract whatsoever with our front-end.

It is the developer's responsibility that this assertion will never be tripped before deploying, because it indicates _a breach of invariants_, which should never, ever happen. It is also important that this is raised as an `Error`, so it trips any build or lint process.

Because this assertion is essentially a sanity check for the developers, _and not an (un)expected exception_, it should not be handled (e.g., by adding a toast or a recovery flow); it should be kept as an assertion.